Fix Windows CMake builds with shared zlib

nyanpasu64 opened the pull request:
On Linux, if shared zlib is present, zlib.h is always available and -lz
links to zlib, even if you don't run find_package(ZLIB).

For some reason I have zlib installed on Windows (possibly from vcpkg),
so find_package(ZLIB) succeeds and ZLIB_FOUND is true.
When Dolphin uses shared zlib on Windows, the problem is that zlib.h
is not in the default include path, and the CMake target is called
ZLIB::ZLIB and there's neither a target nor a library called z.

However, both find_package(ZLIB) and add_subdirectory(Externals/zlib)
create a target called ZLIB::ZLIB, so I'll switch to that instead.
Hopefully this change doesn't break anyone's build.

JosJuice commented:
@dolphin-emu-bot rebuild

Tilka commented on CMakeLists.txt:
```diff
@@ -620,6 +620,7 @@ endif()
 find_package(ZLIB)
 if(ZLIB_FOUND)
   message(STATUS "Using shared zlib")
+  link_libraries(ZLIB::ZLIB)
```
I would have expected that there are more target_link_libraries lines that would need to be changed rather adding this line here.

nyanpasu64 replied:
Oops... I'll remove this and test again at some point, but I'm on Linux at the moment.

nyanpasu64 commented:
Just rebased to master and removed the unnecessary global ZLIB::ZLIB. Building Dolphin works on my branch and fail on master.

I haven't checked if there are any other libraries that can break this way.

JosJuice commented:
@dolphin-emu-bot rebuild

leoetlino reviewed (approved):
sgtm
